### PR TITLE
libm/openlibm: add math library openlibm support.

### PR DESCRIPTION
**Pull request by siryang2006**
add math library openlibm support
(1)open menuconfig  (2)close math.h:build setup->Customize Header Files->math.h (3) select openlibm:Library Rountines->Select math library->Math Library fram openlibm (4)build

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

File: libs/libm/openlibm/Make.defs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+AFLAGS += ${INCDIR_PREFIX}openlibm/openlibm
+AFLAGS += ${INCDIR_PREFIX}openlibm/openlibm/$(ARCH)
+AFLAGS += ${INCDIR_PREFIX}openlibm/openlibm/src
+AFLAGS += ${DEFINE_PREFIX}__BSD_VISIBLE
+
+CSRCS := $(sort $(filter %.c,$(SRCS)))
+ASRCS := $(sort $(filter %.S,$(SRCS)))
+
+# CSRCS and ASRCS have the same name files, we use .S files,so remove them from CSRCS
+CSRCS := $(filter-out $(patsubst %.S,%.c,$(ASRCS)),$(CSRCS))
```
Comment (pkarashchenko) [inline libs/libm/openlibm/Make.defs:116]:
```suggestion
CSRCS := $(filter-out $(patsubst %.S,%.c,$(ASRCS)),$(CSRCS))

```

**Review by pkarashchenko — APPROVED** [review]

File: libs/libm/openlibm/0002-add-math.h-and-complex.h-to-openlibm.patch [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+--- /dev/null
++++ openlibm/openlibm/include/math.h
+@@ -0,0 +1,6 @@
++#ifndef OPENLIBM_MATH_MATH_H
++#define OPENLIBM_MATH_MATH_H
++
++#include <openlibm_math.h>
++
++#endif /* !OPENLIBM_MATH_MATH_H */
+\ No newline at end of file
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply libs/libm/openlibm/0002-add-math.h-and-complex.h-to-openlibm.patch:46]: yes, @siryang2006 please submit patch to openlibm project.

**Review by siryang2006 — COMMENTED** [review]
add code:

#if defined (_WIN32)
 //VBS
typedef __double_t  double_t;
typedef __float_t  float_t;
#elif defined (__NuttX__)
typedef float  float_t;
typedef double double_t;
#endif

File: libs/libm/openlibm/0001-fix-build-float_t-error-float_t-has-not-been-declare.patch [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+--- a/openlibm/openlibm/include/openlibm_math.h
++++ openlibm/openlibm/include/openlibm_math.h
+@@ -149,8 +149,8 @@ extern const union __nan_un {
+     : __signbitl(x))
+ 
+ //VBS
+-//typedef	__double_t	double_t;
+-//typedef	__float_t	float_t;
++typedef float  float_t;
++typedef double double_t;
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply libs/libm/openlibm/0001-fix-build-float_t-error-float_t-has-not-been-declare.patch]: float/double is standard type why need add __NuttX__?

File: libs/libm/openlibm/Make.defs (hunk 1 above)
Comment (siryang2006) [reply libs/libm/openlibm/Make.defs]: resolved

**Pull request merged**